RealmSwift/Tests/MigrationTests.swift

```diff
@@ -1023,6 +1028,31 @@ class MigrationTests: TestCase {
         XCTAssertEqual(try! Realm().objects(SwiftBoolObject.self).count, 10)
     }
 
+    func testMigrationObjectSchema() {
+        let properties: [Property] = autoreleasepool {
+            let realm = try! Realm()
+            return try! realm.write {
+                let obj = SwiftObject()
+                realm.add(obj)
+                return obj.objectSchema.properties
+            }
+        }
+
+        migrateAndTestDefaultRealm { migration, _ in
+            var enumerated = false
+            migration.enumerateObjects(ofType: "SwiftObject") { oldObj, newObj in
+                guard let oldObj = oldObj, let newObj = newObj else {
+                    XCTFail("did not get objects in enumerate")
+                    return
+                }
+                enumerated = true
+                XCTAssertEqual(oldObj.objectSchema.properties, properties)
+                XCTAssertEqual(newObj.objectSchema.properties, properties)
+            }
+            XCTAssertTrue(enumerated)
```

Comment:
Is it worth also testing `migration.oldSchema[type].properties.contains { $0.name == key }` after or before this block too?